### PR TITLE
Run tsc to catch TypeScript syntax error early on

### PR DESCRIPTION
**Pull request by D4nte**
Would have saved me some time :)

### CODE REVIEW

File: api_tests/tsconfig.json [hunk 1]
```diff
@@ -10,5 +10,6 @@
         "noUnusedParameters": true,
         "resolveJsonModule": true,
         "esModuleInterop": true
-    }
+    },
+    "include": ["dry", "btsieve", "e2e", "webgui"]
```
Comment (thomaseizinger) [inline api_tests/tsconfig.json]: :question: Shouldn't need this?

> If the "files" and "include" are both left unspecified, the compiler defaults to including all TypeScript (.ts, .d.ts and .tsx) files in the containing directory and subdirectories

https://www.typescriptlang.org/docs/handbook/tsconfig-json.html

Comment (D4nte) [reply api_tests/tsconfig.json]: Indeed I now realised I previously used `tsc` incorrectly.

File: Makefile.toml [hunk 2]
```diff
@@ -138,6 +139,12 @@ script = [
 '''
 ]
 
+[tasks.check-ts]
+description = "Runs tsc to validate TypeScript syntax."
+workspace = false
+install_script = ["(cd ./api_tests; yarn install; yarn run gen;)"]
```
Comment (thomaseizinger) [inline Makefile.toml]: :mega: 
`yarn run gen` should be in the `postinstall` script I think. Why do we need to repeat it here?

Comment (D4nte) [reply Makefile.toml]: Oh, indeed, it was all messy before! Now cleaned up

File: Makefile.toml [hunk 3]
```diff
@@ -148,7 +155,7 @@ run_task = "ci-flow"
 description = "Install/build dependencies for api tests"
 workspace = false
 private = true
-script = [ "( cd ./api_tests && yarn install; yarn run gen; )" ]
+script = [ "( cd ./api_tests && yarn install;)" ]
```
Comment (da-kami) [inline Makefile.toml]: ❓ The `gen` script was renamed to `postinstall` - but as far as I could see `postinall` is never triggered. Is it triggered implicitly somehow, am I missing something?

Comment (D4nte) [reply Makefile.toml]: It is indeed triggered implicitly after an install

**Pull request merged**